benchmarks/flowertune-llm/evaluation/code/main.py

```diff
@@ -0,0 +1,332 @@
+"""
+This python file is adapted from https://github.com/bigcode-project/bigcode-evaluation-harness/blob/main/main.py
+
+Bigcode evaluation harness (https://github.com/bigcode-project/bigcode-evaluation-harness/tree/main)
+is licensed under the Apache License, Version 2.0.
+
+Citation:
+@misc{bigcode-evaluation-harness,
+  author       = {Ben Allal, Loubna and
+                  Muennighoff, Niklas and
+                  Kumar Umapathi, Logesh and
+                  Lipkin, Ben and
+                  von Werra, Leandro},
+  title = {A framework for the evaluation of code generation models},
+  publisher = {GitHub},
+  journal = {GitHub repository},
+  howpublished = {\url{https://github.com/bigcode-project/bigcode-evaluation-harness}},
+  year = 2022,
+}
+"""
+
+import fnmatch
+import json
+from dataclasses import dataclass, field
+from typing import Optional
+
+import torch
+import transformers
+from accelerate import Accelerator
+from bigcode_eval.evaluator import Evaluator
+from bigcode_eval.tasks import ALL_TASKS
+from transformers import (
+    AutoModelForCausalLM,
+    AutoModelForSeq2SeqLM,
+    AutoTokenizer,
+    HfArgumentParser,
+)
+
+import datasets
+
+
+class MultiChoice:
+    def __init__(self, choices):
+        self.choices = choices
+
+    # Simple wildcard support (linux filename patterns)
+    def __contains__(self, values):
+        for value in values.split(","):
+            if len(fnmatch.filter(self.choices, value)) == 0:
+                return False
+
+        return True
+
+    def __iter__(self):
+        for choice in self.choices:
+            yield choice
+
+
+@dataclass
+class EvalArguments:
+    """Configuration for running the evaluation."""
+
+    prefix: Optional[str] = field(
+        default="",
+        metadata={
+            "help": "Prefix to add to the prompt. For example InCoder needs prefix='<| file ext=.py |>\n'"
+        },
+    )
+    do_sample: Optional[bool] = field(
+        default=True,
+        metadata={"help": "Sample from the language model's output distribution."},
+    )
+    temperature: Optional[float] = field(
+        default=0.2, metadata={"help": "Sampling temperature used for generation."}
+    )
+    top_k: Optional[int] = field(
+        default=0, metadata={"help": "Top-k parameter used for generation."}
+    )
+    top_p: Optional[float] = field(
+        default=0.95, metadata={"help": "Top-p parameter used for nucleus sampling."}
+    )
+    n_samples: Optional[int] = field(
+        default=1,
+        metadata={"help": "Number of completions to generate for each sample."},
+    )
+    eos: Optional[str] = field(
+        default="<|endoftext|>", metadata={"help": "end of sentence token."}
+    )
+    seed: Optional[int] = field(
+        default=2024, metadata={"help": "Random seed used for evaluation."}
+    )
+
+
+def parse_args():
```

Comment:
@yan-gao-GY , the `README.md` only sets 7 input arguments but here the code defines 23 of them. I wonder if it's better to remove those that aren't supposed to be changed by a person running the evaluation. Better leave them hardcoded in the file? (same would apply to `--save_generations` `--save_references` which should be passed all the time)